fuse_models/include/fuse_models/parameters/odometry_2d_publisher_params.h

```diff
@@ -69,7 +73,23 @@ struct Odometry2DPublisherParams : public ParameterBase
   {
     nh.getParam("publish_tf", publish_tf);
     nh.getParam("predict_to_current_time", predict_to_current_time);
-    nh.getParam("tf_publish_frequency", tf_publish_frequency);
+    nh.getParam("publish_frequency", publish_frequency);
+
+    std::vector<double> process_noise_diagonal(8, 0.0);
+    nh.param("process_noise_diagonal", process_noise_diagonal, process_noise_diagonal);
+
+    if (process_noise_diagonal.size() != 8)
+    {
+      throw std::runtime_error("Process noise diagonal must be of length 8!");
+    }
+
+    if (std::any_of(process_noise_diagonal.begin(), process_noise_diagonal.end(),
+                    [](const auto& v) { return v < 0.0; }))  // NOLINT(whitespace/braces)
```

Comment:
```suggestion
                    [](const auto& v) { return v <= 0.0; }))  // NOLINT(whitespace/braces)
```
You're probably duplicating code I wrote elsewhere, 0's really shouldn't be allowed either.

Reply:
If I don't allow `0.0`, then the default value for `process_noise_diagonal` can't be all `0.0`s as it's now. Then the question is what value should I use in that case. Maybe something like `1e-6`? But the diagonal represents different state components, with different units, so some should be smaller than others, like the velocities or accelerations. :thinking: 

The reason I allowed `0.0`s in this case is because it's the process noise covariance, so if the user doesn't specify it, then it'd be like there's no process noise, that it's not recommended, but the math won't blow up. Maybe I can add a `ROS_WARN` saying that it's not recommended to have zeros in the process noise covariance diagonal.

Regarding the check, I can only find this in unicycle 2D motion model, but it doesn't perform the check I do here. Maybe it should also have a check similar to mine.
https://github.com/locusrobotics/fuse/blob/9933456ecc24ba9b649a8a2885be3f852306efee/fuse_models/src/unicycle_2d.cpp#L99-L105

Reply:
I expect the math will blow up actually. Inside the motion model cost function, the error is normalized by the square root information matrix. If there are any zero terms in the process noise, the covariance cannot be inverted into an information matrix.

Reply:
Exactly, the `process_noise_diagonal` set here is NOT used in: https://github.com/locusrobotics/fuse/blob/devel/fuse_models/src/unicycle_2d.cpp#L246 and https://github.com/locusrobotics/fuse/blob/devel/fuse_models/src/unicycle_2d_state_kinematic_constraint.cpp#L81, where a covariance with zeros will certainly blow up.

This one is only used in https://github.com/locusrobotics/fuse/pull/109/files#diff-ec05e4b7290a6166c67ff2570f66529fR320, that will simply make that addition a nop.

That being said, if we agree on some initial values, or some WARN messages, I could add that, so the user is aware of a potentially bad configuration.

Ideally, the publisher should be using the same process noise covariance as the unicycle_2d motion model. Indeed, I think the publisher should allow the user to specify the motion model to use for prediction, but that would be another story, for another PR.